core/trino-spi/src/main/java/io/trino/spi/block/EncoderUtil.java

```diff
@@ -120,4 +134,156 @@ public static byte[] retrieveNullBits(SliceInput sliceInput, int positionCount)
             throw new UncheckedIOException(e);
         }
     }
+
+    static void compactBytesWithNullsVectorized(SliceOutput sliceOutput, byte[] values, boolean[] isNull, int offset, int length)
+    {
+        requireNonNull(sliceOutput, "sliceOutput is null");
+        checkFromIndexSize(offset, length, values.length);
```

Comment:
This check is redundant with `checkFromIndexSize(rawOffset, positionCount, rawValues.length)` already happening in BlockEncoding implementations. We can drop these checks.

Reply:
I’m not sure that’s the right strategy here. Even though this check should never fail as a result of earlier checks in the block encoding implementations, the JIT needs to “see” the bounds checks happening within the current method scope after inlining. Otherwise, it’s going to have to intersperse bounds checks within the loop body which will hurt performance. If everything gets inlined into the block encoder, then we’ll be ok- but if the compact methods aren’t inlined then this check is load bearing.

Honestly, we could go the other way on this and inline the implementation of checkFromIndexSize into these methods so that we’re not depending on that method getting inlined into this method.